### PR TITLE
Added unit tests for `boa_ast::Punctuator`

### PR DESCRIPTION
**Pull request by Razican**
This is a follow up of #2883, which adds more tests for `boa_ast::Punctuator`.

### CODE REVIEW

File: boa_ast/src/punctuator/mod.rs [hunk 1]
```diff
@@ -177,6 +177,7 @@ impl Punctuator {
             Self::Mul => Some(BinaryOp::Arithmetic(ArithmeticOp::Mul)),
             Self::Div => Some(BinaryOp::Arithmetic(ArithmeticOp::Div)),
             Self::Mod => Some(BinaryOp::Arithmetic(ArithmeticOp::Mod)),
+            Self::Exp => Some(BinaryOp::Arithmetic(ArithmeticOp::Exp)),
```
Comment (Razican) [inline boa_ast/src/punctuator/mod.rs:180]: This was somehow missing before, but it doesn't seem to fix any new test.

**Review by HalidOdat — APPROVED** [review]
Looks good to me!

**Comment by Razican** [comment]
Prefix removed :)

**Review by HalidOdat — APPROVED** [review]

**Review by jedel1043 — APPROVED** [review]
LGTM!

**Pull request merged**